### ISSUE TITLE
Angular 8 Support

### ISSUE THREAD

**Issue by internalsystemerror** [ngxs/store#1125]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => https://github.com/ngxs/store/blob/master/CONTRIBUTING.md
[ ] Other... Please describe:
</code></pre>

A statement as to the current status regarding Angular 8 and Universal would be appreciated as I can't find one if this exists. I can see various dependency update chores however I can see the benefit of having this ticket closed when full support is reached for anyone else finding this.

**Comment by splincode** [comment]
Thank you for reporting

**Comment by arturovt** [comment]
@internalsystemerror 
We know about that and we're working on that, actually @splincode already has a skeleton for upgrading to Angular 8. It's currently working with Angular 8 (I checked), but not with Ivy (it's related to `@angular/cli`, not our issue, will be fixed in 2-3 weeks as Google team says)

**Comment by internalsystemerror** [comment]
@arturovt thanks for the feedback. Just planning our own upgrades and getting the status of various libraries. A ticket like this I feel is just good for reference.

**Comment by jtomek** [comment]
> @internalsystemerror
> We know about that and we're working on that, actually @splincode already has a skeleton for upgrading to Angular 8. It's currently working with Angular 8 (I checked), but not with Ivy (it's related to `@angular/cli`, not our issue, will be fixed in 2-3 weeks as Google team says)

Can you please publish the version of ngxs that works with Angular 8 (without Ivy support)? I just went through a lengthy upgrade from 7 -> 8 and just now noticed that ngxs isn't supporting Angular 8.

**Comment by arturovt** [comment]
@jtomek 
This is the third place where you’re telling that something doesn’t work. Create an issue with reproduce!

**Comment by jtomek** [comment]
> @jtomek
> This is the third place where you’re telling that something doesn’t work. Create an issue with reproduce, please!

As people have reported, it's not working with Angular 8. I don't think it's necessary to create another issue regarding this topic. 

Maybe I should create one about the future of ngxs. I don't want to sound bitter, but I think it's a big problem that so many people are left hanging while everybody package had such a long runway to prepare for a new version. I've implemented ngxs as an integral part of my app because it's been so easy and reduced the boilerplate. But as pointed out in a different discussion here, ngxs has been stale for some time and this problem with Angular 8 is just not helping at all. 

I am wondering, I saw that you worked on ngxs examples. I think that the whole community can only thank you for that. But why prioritize working on minor examples around over such as important topic that a new Angular version is? It's not only me but also others who I am sure will have to consider or will be forced by their organizations to switch to ngrx.

**Comment by splincode** [comment]
@jtomek My project uses Angular 8 and everything works, except for warnings when installing npm you will not see anything that could not work with NGXS

**Comment by splincode** [comment]
@jtomek 

> will be forced by their organizations to switch to ngrx.

up to you

**Comment by jtomek** [comment]
> @jtomek My project uses Angular 8 and everything works, except for warnings when installing npm you will not see anything that could not work with NGXS

I see. Are you using latest release of NGXS v3.4.3?

**Comment by splincode** [comment]
yes, I use v3.4.3

**Comment by arturovt** [comment]
@splincode I agree with you, I'm also using NGXS with Angular 8 (stable) and it works perfectly!

**Comment by jtomek** [comment]
I thought that the latest version of NGXS was breaking my code and I started reversing my upgrade. Thank you @splincode and @arturovt for the information that it was working under Angular 8. I've now managed to get it working (there is a new polyfill to be included and loadChildren was and then wasn' upgraded in my case).

**Comment by arturovt** [comment]
@jtomek
I’m happy to help you and my purpose is to make our user’s development more productive.
Just if you’re asking anything - would be great if you show us an error or create an issue, really hard to determine what’s really going wrong there :wink:

**Comment by xmlking** [comment]
This project use ngxs heavenly and it works with angular 8
https://github.com/xmlking/ngx-starter-kit

**Comment by andrewjrhill** [comment]
Is there an ETA on Ivy support? Thanks.

**Comment by arturovt** [comment]
@andrewjrhill I've told about Ivy.

**Comment by andrewjrhill** [comment]
@arturovt - missed that; thanks!

**Comment by lppedd** [comment]
Just to confirm - works fine under 8 (and _next_) without Ivy :)

**Comment by arturovt** [comment]
@lppedd thanks for confirming.

Gentlemen, to make things clear - we can't focus on Ivy, as it's senseless right now. Igor Minar said that Ivy will be stable in Angular 9 or 10.

**Comment by internalsystemerror** [comment]
Makes sense to me. I just don't want to go through the effort of upgrading everything if it means our builds are going to fail due to incorrect warnings. This is my intention behind this ticket.

**Comment by arturovt** [comment]
Ivy is not compatible with ng-packagr right now, that's used with `@angular-devkit`. If Google makes it compatible - we also can start the integration with Ivy.

**Comment by Koslun** [comment]
@arturovt 
> Igor Minar said that Ivy will be stable in Angular 9 or 10.

Do you have a source or know when he said this? And do you mean stable in the sense of API stability or just that it'll be the default?

During 2019 the ng-conf keynotes and a few other talks this year the message I heard was that Ivy would be an opt-in preview in Angular 8 but default in Angular 9. With Angular 8 released they did seem to keep that deadline.

With 97% of tests in Google already passing and virtually all members of the Angular team working on this I would expect them to keep the deadline of Angular 9 with some risk that it'll be delayed until Angular 10 if they can't make it stable enough.

So think we should plan on Ivy coming in Angular 9 and try to test it with Ivy as soon as possible so that we can:
1. Track how close this library is towards working with Ivy. This would also be helpful for users of the library as we'd be more aware of how realistic it will be to turn on Ivy in the near-future.
2. Report unintentional breaking changes and bugs for Ivy. This way hopefully the Angular team will solve these issues before Ivy becomes default.

I do expect the main changes in this library to integrate Ivy to not really happen until a beta or release candidate of Angular 9 is released and up until then mainly reporting the issues we face with Ivy.

**Comment by arturovt** [comment]
@Koslun just pasting

`it's not fully functional (opt-in preview) and as suggested by Igor Minar in ngConf 2019, View Engine is still recommended for new apps.`

**Comment by mnahkies** [comment]
@arturovt I inquired about this over here https://github.com/ng-packagr/ng-packagr/issues/1175#issuecomment-505122191 and it sounds like angular packages produced with the current version of `ng-packagr` should be consumable by both View Engine and Ivy applications

**Comment by arturovt** [comment]
@mnahkies 
That's great. But actually it was blocked by [this PR](https://github.com/angular/angular/pull/30591) and as you see it was merged 6 days ago.

**Comment by Carniatto** [comment]
Any news on this?

**Comment by splincode** [comment]
@markwhitfeld ping

**Comment by lppedd** [comment]
@Carniatto what's the issue? I'm using Angular 8 in production since it was in Beta, nothing at all for me.

**Comment by Carniatto** [comment]
There is a lot of warnings when I update angular to 8 cuz the peerDependencies for NGXS are wrong. Based on this I found this thread and checked the changelogs that does not say a work about support to Angular 8. If is compatible why the required Angular version on package.json is still angular 7.2.15?

**Comment by lppedd** [comment]
@Carniatto Ahhh okay. Well when I updated, I reviewed warnings one by one. They're just warnings, you can run libs safely anyway, most of the times.

I suppose they're waiting for Ivy support (?).

**Comment by Carniatto** [comment]
@lppedd Thanks for the info, that's was my question is I could just ignore the warnings without side effects, we use ngxs in a large product and I wanted to be safe

**Comment by lppedd** [comment]
@Carniatto Update, give a `dev` spin of the product and check.
Everything should work as is, without Ivy obviously.

**Comment by arturovt** [comment]
Gentlemen, created [an issue](https://github.com/angular/angular/issues/31791) related to Ivy.

**Comment by markwhitfeld** [comment]
The Ivy issue will be tracked locally here: #1165

**Comment by splincode** [comment]
Released in 3.5.0
